source/common/quic/envoy_quic_server_connection.h

```diff
@@ -134,6 +149,7 @@ class EnvoyQuicServerConnection : public quic::QuicConnection, public QuicNetwor
   void OnCanWrite() override;
 
   bool actuallyDeferSend() const { return defer_send_in_response_to_packets(); }
+  void OnFirstPacketReceived(const quic::QuicReceivedPacket& packet);
```

Comment:
Can you override QuicConnection::ProcessUdpPacket() directly instead of plumbing through the session?

Reply:
Good idea, thanks. Done.